// app/request-sender/index.js
/*var config = require('./../config');*/

/*const BACKEND_HOST = config.get('backendHost');*/
const BACKEND_HOST = 'http://localhost:8002';

function make_base_auth_token(userName, password) {
    var token = userName + ':' + password;
    var hash = btoa(token);
    return "Basic " + hash;
}

function basicAuthGET(url, token, successFunction, errorFunction) {
    var result;
    $.ajax({
        type: 'GET',
        dataType: 'json',
        contentType: "application/json; charset=utf-8",
        url: BACKEND_HOST + url,
        beforeSend: function (xhr) {
            xhr.setRequestHeader('Authorization', token);//method sets the value of an HTTP request header
        },
        success: function (data) {
            successFunction(data);
            result = data;
        },
        error: function (xhr, status, err) {
            errorFunction(xhr, status, err);
        }
    })
    return result;
};

function basicAuthPOST(url, token, dataJson, successFunction, errorFunction) {
    var result;
    $.ajax({
        type: 'POST',
        url: BACKEND_HOST + url,
        dataType: 'json',
        contentType: "application/json; charset=utf-8",
        data: JSON.stringify(dataJson),
        beforeSend: function (xhr) {
            xhr.setRequestHeader('Authorization', token);
        },
        success: function (data) {
            successFunction(data);
        },
        error: function (xhr, status, err) {
            errorFunction(xhr, status, err);
        }
    });
    return result;
};

function GET(url, successFunction, errorFunction) {
    var result;
    $.ajax({
        type: 'GET',
        dataType: 'json',
        contentType: "application/json; charset=utf-8",
        url: BACKEND_HOST + url,
        success: function (data) {
            successFunction(data);
            result = data;
        },
        error: function (xhr, status, err) {
            errorFunction(xhr, status, err);
        }
    })
    return result;
};

function POST(url, dataJson, successFunction, errorFunction) {
    var result;
    $.ajax({
        type: 'POST',
        url: BACKEND_HOST + url,
        dataType: 'json',
        contentType: "application/json; charset=utf-8",
        data: JSON.stringify(dataJson),
        success: function (data) {
            successFunction(dataJson);
        },
        error: function (xhr, status, err) {
            errorFunction(xhr, status, err);
        }
    });
    return result;
};

module.exports.make_base_auth_token = make_base_auth_token;
module.exports.basicAuthGET = basicAuthGET;
module.exports.basicAuthPOST = basicAuthPOST;
module.exports.GET = GET;
module.exports.POST = POST;
